src/Bridge/Doctrine/Orm/CollectionDataProvider.php

```diff
@@ -67,6 +68,9 @@ public function getCollection(string $resourceClass, string $operationName = nul
             }
         }
 
-        return $queryBuilder->getQuery()->getResult();
+        $query = $queryBuilder->getQuery();
+        //forces doctrine to not lazy load relations, see http://stackoverflow.com/questions/9848747/primary-key-of-owning-side-as-a-join-column
+        $query->setHint(Query::HINT_FORCE_PARTIAL_LOAD, true);
```

Comment:
I think that this might give unwanted results, so I was wondering what solutions we might have. 
Maybe that just documenting it would be enough, but as this is only modifiable on the `Query` object (not the `QueryBuilder`), we may want to add a easy way to set the option. WDYT?